design/update.go

```diff
@@ -0,0 +1,68 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var updateInfo = a.Type("UpdateInfo", func() {
+	a.Description(`JSONAPI for the update info object. See also http://jsonapi.org/format/#document-resource-object`)
+	a.Attribute("status", d.String, "The update status", func() {
+	})
+	a.Attribute("last-time-updated", d.DateTime, "When an update of the last batch of tenants was finished", func() {
+		a.Example("2016-11-29T23:18:14Z")
+	})
+	a.Attribute("failed-count", d.Integer, "The number of failed tenant updates", func() {
+	})
+	a.Attribute("file-versions", a.ArrayOf(fileWithVersion), "Lis of files and their versions used for the last finished run", func() {
+	})
+	a.Attribute("links", genericLinks)
+})
+
+var fileWithVersion = a.Type("FileWithVersion", func() {
+	a.Attribute("file-name", d.String, "Name of a file")
+	a.Attribute("version", d.String, "Version of the file that was set when the last update was finished")
+})
+
+var _ = a.Resource("update", func() {
+	a.BasePath("/api/update")
+	a.Action("start", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.POST(""),
+		)
+		a.Params(func() {
+			a.Param("cluster_url", d.String, "the URL of the OSO cluster the update should be limited to")
+			a.Param("env_type", d.String, "environment type the update should be executed for")
```

Comment:
Maybe it's worth adding a.Enum(...) to list all valid type here?

Reply:
good point  - added here 1746454

Reply:
and here 8ac68a5 
the original commit is adding an enum only for status